internal/pkg/xray/tree.go

```diff
@@ -62,15 +67,60 @@ func buildTree(traceMap map[string]*Trace, sourceTrace *Trace) (*Tree, error) {
 
 	// Find all the leaf nodes and their complete paths using DFS algorithm
 	var singlePath []*Segment
+
+	linkedTraceIds := maps.Keys(linkedTraceToSegment)
+	linkedTraceLimitExceeded := false
+
+	//get all linked traces in one call
+	if fetchLinkedTraces && len(linkedTraceIds) > 0 && *depth < MAX_TREE_DEPTH {
+		linkedTraceMap, err := opts.getTraces(ctx, opts.xrayClient, linkedTraceIds)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch linked traces %s with error: %w", linkedTraceIds, err)
+		}
+		*depth = *depth + 1
+		for linkedTraceId, linkedTrace := range linkedTraceMap {
+			linkedTree, err := buildTree(linkedTraceMap, linkedTrace, fetchLinkedTraces, ctx, opts, depth)
+			if err != nil {
+				return nil, fmt.Errorf("failed to build trace tree %s with error: %w", linkedTraceId, err)
+			}
+
+			parentSegment := linkedTraceToSegment[*linkedTree.Root.TraceId]
+			parentSegment.children = append(parentSegment.children, linkedTree.Root)
+		}
+	}
+	if *depth >= MAX_TREE_DEPTH {
+		linkedTraceLimitExceeded = true
+	}
 	leafPaths := FindLeafSegmentPaths(treeRootSegment, singlePath)
 
 	return &Tree{
-		Root:        treeRootSegment,
-		Paths:       leafPaths,
-		SourceTrace: sourceTrace,
+		Root:                     treeRootSegment,
+		Paths:                    leafPaths,
+		SourceTrace:              sourceTrace,
+		LinkedTraceLimitExceeded: linkedTraceLimitExceeded,
 	}, nil
 }
 
+// check if there are any linked traces associated with the segment and add them to the map
+func getLinkedTraces(segment *Segment, linksToSegMap map[string]*Segment) {
+	linkedTraces := segment.Links
+	for _, trace := range linkedTraces {
```

Comment:
Something I had noticed when I was looking into linked traces was that `child` links were only found under `subsegments` and `parent` links were found under a `Segment`. I did not find anything in the [docs](https://docs.aws.amazon.com/xray/latest/devguide/xray-tracelinking.html) mentioning where the links are found. But I think we can just keep this logic as there could be an example where this assumption is False.

Reply:
Yea i noticed the same thing but figured that theres probably a reason that each segment has a `Links` parameter in it